File: src/Lua/1987.lua
while (true) do
    x, y = io.read()
    if x == nil then break end

    sum = 0
    for i = 1, #y do
        sum = sum + tonumber(y[i])
    end

    if y % 3 == 0 then print(sum .. " sim") 
    else print(sum .. " nao") end
end

File: src/Cpp/1987.cpp
#include <iostream>

using namespace std;

int main() {
    int n, soma = 0;
    string num;

    while(cin >> n >> num) {

        for (int i = 0; i < n; i++) {
            soma += (int)num[i]-48;
        }

        if (soma % 3 == 0) cout << soma << " sim" << endl;
        else cout << soma << " nao" << endl;

        soma = 0;
    }

    return EXIT_SUCCESS;
}